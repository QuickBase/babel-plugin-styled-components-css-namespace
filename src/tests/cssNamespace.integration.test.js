import path from 'path';
import React from 'react';
import renderer from 'react-test-renderer';
import { transformFileSync } from 'babel-core';
import 'jest-styled-components';

const evalFixture = filename => {
  const { code } = transformFileSync(filename, {
    plugins: [
      [
        path.join(__dirname, '../index.js'),
        { rawCssNamespace: ['#different-wrapper'] }
      ]
    ]
  });

  if (code == null) throw new Error(`Fixture not found: ${filename}`);

  return eval(code);
};

describe('styled-components output', () => {
  test('for a style block with no selectors', () => {
    const Simple = evalFixture(
      path.join(__dirname, 'fixtures/integration/simple.js')
    );
    expect(
      renderer.create(<Simple backgroundColor="#333" />).toJSON()
    ).toMatchSnapshot();
  });

  test('for a style block with &&', () => {
    const Input = evalFixture(
      path.join(__dirname, 'fixtures/integration/double_ampersand.js')
    );
    expect(
      renderer
        .create(<Input borderWidth="1px" borderColor="fuchsia" />)
        .toJSON()
    ).toMatchSnapshot();
  });

  test('for a style block with a sibling selector', () => {
    const Button = evalFixture(
      path.join(__dirname, 'fixtures/integration/sibling_selector.js')
    );
    expect(
      renderer.create(<Button padding="4px" spaceBetween="8px" />).toJSON()
    ).toMatchSnapshot();
  });

  // ENDLESS LOOP
  // test('for a style block with interpolated selectors', () => {
  //   const Parent = evalFixture(
  //     path.join(__dirname, 'fixtures/interpolated_selector.js')
  //   );
  //   expect(
  //     renderer
  //       .create(
  //         <Parent childStyles="transform: scale(90%);" spaceBetween="12px" />
  //       )
  //       .toJSON()
  //   ).toMatchSnapshot();
  // });
});
