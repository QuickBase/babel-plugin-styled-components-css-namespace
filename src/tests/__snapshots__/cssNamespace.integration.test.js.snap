// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styled-components output for a style block with && 1`] = `
#different-wrapper .c0.c0 {
  border: 1px solid fuchsia;
}

<input
  className="c0"
/>
`;

exports[`styled-components output for a style block with a sibling selector 1`] = `
#different-wrapper .c1 + .c0 {
  background-color: 'yellow';
}

<div
  className="c0 c1"
/>
`;

exports[`styled-components output for a style block with interpolated selectors 1`] = `
#different-wrapper .c0 {
  position: relative;
}

#different-wrapper .c0 .c1 {
  -webkit-transform: scale(90%);
  -ms-transform: scale(90%);
  transform: scale(90%);
}

#different-wrapper .c2 + .c0,
#different-wrapper .c0 + .sc-dnqmqq {
  margin-right: 12px;
}

<div
  className="c0"
/>
`;

exports[`styled-components output for a style block with no selectors 1`] = `
#different-wrapper .c0 {
  background-color: 'yellow';
}

<div
  className="c0"
/>
`;
